go/adbc/sqldriver/driver_internals_test.go

```diff
@@ -59,3 +65,130 @@ func TestParseConnectStr(t *testing.T) {
 		assert.Equal(t, expectOpts, gotOpts)
 	}
 }
+
+var (
+	tz       = time.FixedZone("North Idaho", -int((8 * time.Hour).Seconds()))
+	testTime = time.Date(2023, time.January, 26, 15, 40, 39, 123456789, tz)
+)
+
+func TestNextRowTypes(t *testing.T) {
+	tests := []struct {
+		arrowType      arrow.DataType
+		arrowValueFunc func(*testing.T, array.Builder)
+		golangValue    any
+	}{
+		{
+			arrowType: &arrow.StringType{},
+			arrowValueFunc: func(t *testing.T, b array.Builder) {
+				t.Helper()
+				b.(*array.StringBuilder).Append("my-string")
+			},
+			golangValue: "my-string",
+		},
+		{
+			arrowType: &arrow.Date32Type{},
+			arrowValueFunc: func(t *testing.T, b array.Builder) {
+				t.Helper()
+				b.(*array.Date32Builder).Append(arrow.Date32FromTime(testTime))
+			},
+			golangValue: testTime.UTC().Truncate(24 * time.Hour),
+		},
+		{
+			arrowType: &arrow.Date64Type{},
+			arrowValueFunc: func(t *testing.T, b array.Builder) {
+				t.Helper()
+				b.(*array.Date64Builder).Append(arrow.Date64FromTime(testTime))
+			},
+			//TODO type DATE64 is "milliseconds since the UNIX epoch", but arrow.Date64FromTime() truncates at 24h
+			// https://arrow.apache.org/docs/cpp/api/datatype.html#_CPPv4N5arrow4Type4type6DATE64E
+			//golangValue: testTime.UTC().Truncate(time.Millisecond),
```

Comment:
Is this just a weird detail of Arrow? If Date64 is "milliseconds since the UNIX epoch", then why truncate at 24 hours?

Reply:
It's still a date, IIRC this exists for easy compatibility with some systems that store dates in this format (most notably, Java)